### PR TITLE
[BUG #5268] Fix bug when registering with Eureka through EurekaInstanceRegisterRepository#persistInstance.

### PR DESCRIPTION
**Pull request by ShawnJim**
ref #5268.

### CODE REVIEW

File: shenyu-registry/shenyu-registry-eureka/src/main/java/org/apache/shenyu/registry/eureka/EurekaInstanceRegisterRepository.java [hunk 1]
```diff
@@ -50,36 +50,120 @@ public class EurekaInstanceRegisterRepository implements ShenyuInstanceRegisterR
...
+        if (vipAddressResolver == null) {
+            vipAddressResolver = new Archaius1VipAddressResolver();
+        }
+
+        // Builder the instance information to be registered with eureka server
+        final InstanceInfo.Builder builder = InstanceInfo.Builder.newBuilder(vipAddressResolver);
+
+        // set the appropriate id for the InstanceInfo, falling back to datacenter Id if applicable, else hostname
+        String instanceId = eurekaInstanceConfig.getInstanceId();
+        if (instanceId == null || instanceId.isEmpty()) {
```
Comment (moremind) [inline shenyu-registry/shenyu-registry-eureka/src/main/java/org/apache/shenyu/registry/eureka/EurekaInstanceRegisterRepository.java:105]: use StringUtils.isEmpty

File: shenyu-registry/shenyu-registry-eureka/src/main/java/org/apache/shenyu/registry/eureka/EurekaInstanceRegisterRepository.java [hunk 2]
```diff
@@ -50,36 +50,120 @@ public class EurekaInstanceRegisterRepository implements ShenyuInstanceRegisterR
...
+        String defaultAddress;
+        if (eurekaInstanceConfig instanceof RefreshableInstanceConfig) {
+            // Refresh AWS data center info, and return up to date address
+            defaultAddress = ((RefreshableInstanceConfig) eurekaInstanceConfig).resolveDefaultAddress(false);
+        } else {
+            defaultAddress = eurekaInstanceConfig.getHostName(false);
+        }
+
+        // fail safe
+        if (defaultAddress == null || defaultAddress.isEmpty()) {
```
Comment (moremind) [inline shenyu-registry/shenyu-registry-eureka/src/main/java/org/apache/shenyu/registry/eureka/EurekaInstanceRegisterRepository.java:123]: same

**Review by moremind — CHANGES_REQUESTED** [review]

File: shenyu-registry/shenyu-registry-eureka/src/main/java/org/apache/shenyu/registry/eureka/EurekaInstanceRegisterRepository.java (hunk 1 above)
Comment (ShawnJim) [reply shenyu-registry/shenyu-registry-eureka/src/main/java/org/apache/shenyu/registry/eureka/EurekaInstanceRegisterRepository.java]: It`s done, thank you for review. :)

File: shenyu-registry/shenyu-registry-eureka/src/main/java/org/apache/shenyu/registry/eureka/EurekaInstanceRegisterRepository.java (hunk 2 above)
Comment (ShawnJim) [reply shenyu-registry/shenyu-registry-eureka/src/main/java/org/apache/shenyu/registry/eureka/EurekaInstanceRegisterRepository.java]: done.

**Review by moremind — APPROVED** [review]

**Pull request merged**